src/main/java/com/krishagni/openspecimen/msk2/importer/CarsImporter.java

```diff
@@ -0,0 +1,278 @@
+		
+		CollectionProtocolDetail cpDetail = checkCp(event, detail.getObject());
+		checkEvent(cpDetail, event);
+		checkSr(event);
+		
+		return null;
+	}
+
+	private CollectionProtocolEventDetail toEvent(CarsDetail detail) {
+		CollectionProtocolEventDetail eventDetail = new CollectionProtocolEventDetail();
+		SpecimenRequirementDetail srDetail = new SpecimenRequirementDetail();
+		
+		eventDetail.setCpShortTitle(detail.getIrbNumber());
+		eventDetail.setEventLabel(detail.getCycleName() + detail.getTimepointName());
+		
+		srDetail.setCpShortTitle(detail.getIrbNumber());
+		srDetail.setType(detail.getSpecimenType());
+		srDetail.setStorageType(SPECIMEN_STORAGE_TYPE_VIRTUAL);
+		srDetail.setAnatomicSite(NOT_SPECIFIED);
+		srDetail.setLaterality(NOT_SPECIFIED);
+		srDetail.setPathology(NOT_SPECIFIED);
+		srDetail.setCollectionProcedure(NOT_SPECIFIED);
+		srDetail.setEventLabel(detail.getCycleName() + detail.getTimepointName());
+		srDetail.setName(detail.getProcedureName());
+		srDetail.setInitialQty(new BigDecimal(0));
+		srDetail.setCollectionContainer(detail.getCollectionContainer());
+		srDetail.setSpecimenClass(getSpecimenClass(detail.getSpecimenType()));
+		
+		eventDetail.setSpecimenRequirements(Arrays.asList(srDetail));
+		
+		return eventDetail;
+	}
+
+	private String getSpecimenClass(String specimenType) {
+		ListPvCriteria crit = new ListPvCriteria()
+				.includeParentValue(true)
+				.parentAttribute("specimen_type")
+				.values(Collections.singletonList(specimenType));
+		
+		ResponseEvent<List<PvDetail>> resp = pvSvc.getPermissibleValues(request(crit));
+		List<PvDetail> payload = resp.getPayload();
+		
+		return payload.isEmpty() ? "" : payload.iterator().next().getParentValue();
+	}
+
+	private <T> RequestEvent<T> request(T payload) {
+		return new RequestEvent<T>(payload);
+	}
+	
+	//////////////////////////
+	//
+	// Specimen Requirement
+	//
+	/////////////////////////
+	
+	private void checkSr(CollectionProtocolEventDetail event) throws Exception {
+		SpecimenRequirementDetail srDetail = srMap.get(event.getSpecimenRequirements().iterator().next().getName());
+		
+		if (srDetail == null) {
+			SpecimenRequirement srFromDb = getSrFromDb(event);
+			srDetail = srFromDb != null ? SpecimenRequirementDetail.from(srFromDb) : null;
+		}
+		
+		if (srDetail != null) {
+			updateSr(event.getSpecimenRequirements(), srDetail);
+		} else {
+			srDetail = createSr(event.getSpecimenRequirements());
+		}
+		
+		srMap.put(event.getSpecimenRequirements().iterator().next().getName(), srDetail);
+	}
+	
+	private SpecimenRequirement getSrFromDb(CollectionProtocolEventDetail event) {
+		String srName = event.getSpecimenRequirements().iterator().next().getName();
+		CollectionProtocolEvent cpe = daoFactory
+				.getCollectionProtocolDao()
+				.getCpeByShortTitleAndEventLabel(event.getCpShortTitle(), event.getEventLabel());
+		
+		Set<SpecimenRequirement> specimenRequirements = cpe != null ? cpe.getSpecimenRequirements() : Collections.emptySet();
```

Comment:
Get only top-level requirements. You will getting information about only the primary specimens.